Source/Core/Core/PowerPC/PowerPC.cpp

```diff
@@ -126,19 +127,8 @@ static void ResetRegisters()
   SystemTimers::DecrementerSet();
 }
 
-void Init(int cpu_core)
+static void ResetTLB()
 {
-  // NOTE: This function runs on EmuThread, not the CPU Thread.
-  //   Changing the rounding mode has a limited effect.
-  FPURoundMode::SetPrecisionMode(FPURoundMode::PREC_53);
-
-  s_invalidate_cache_thread_safe =
-      CoreTiming::RegisterEvent("invalidateEmulatedCache", InvalidateCacheThreadSafe);
-
-  memset(ppcState.sr, 0, sizeof(ppcState.sr));
-  ppcState.pagetable_base = 0;
-  ppcState.pagetable_hashmask = 0;
-
   for (int tlb = 0; tlb < 2; tlb++)
```

Comment:
It can also utilize generate, since it's resetting to a fixed state, which can be as follows:

```cpp
for (auto& tlb : ppcState.tlb)
{
  std::generate(std::begin(tlb), std::end(tlb), [] {
    return tlb_entry{{TLB_TAG_INVALID, TLB_TAG_INVALID}, {0, 0}, {0, 0}, 0};
  });
}
```

or it could be done using fill:

```cpp
static constexpr tlb_entry clean{{TLB_TAG_INVALID, TLB_TAG_INVALID}, {0, 0}, {0, 0}, 0};
  
for (auto& tlb : ppcState.tlb)
{
  std::fill(std::begin(tlb), std::end(tlb), clean);
}
```

Would you prefer either of those, or just ranged for loops?